### PR TITLE
Autogenerate values

### PR DESCRIPTION
**Pull request by ereslibre**
Auto generate chart values based on common values.

Also, before releasing a chart, ensure that there are no differences with the checked in `values.yaml`. This allows us to check `values.yaml` in the tree and ensure that it's up to date before trying to release a new version of the Helm charts.

Matt's feedback: https://github.com/kubewarden/helm-charts/issues/76#issuecomment-1090591393

### CODE REVIEW

File: .github/workflows/helm-chart-release.yml [hunk 1]
```diff
@@ -48,6 +48,10 @@ jobs:
           git config user.name "$GITHUB_ACTOR"
           git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
 
+      - name: Check Helm generated values are up-to-date before releasing
```
Comment (viccuad) [inline .github/workflows/helm-chart-release.yml:51]: I would like to also run this for PRs in an additional workflow.

**Review by viccuad — COMMENTED** [review]
With this approach, we have 3 values, one per chart, and the correct default values from the `common-values.yml`.  I would prefer to only have need 1 values.yml file for all charts, though.

**Comment by ereslibre** [comment]
> With this approach, we have 3 values, one per chart

The idea was that you can have a single values.yaml that you apply to all charts. The chart being applied only consumes part of it.

I can add top-level keys for each chart if you prefer, as:

```
common:
  ...
kubewarden-controller:
  ...
```

So the single `values.yaml` is even more explicit. I wanted to avoid artificial extra key-values that make overriding values more verbose. It's also true that having this extra key-values prevents possible conflicts.

**Comment by viccuad** [comment]
> The idea was that you can have a single values.yaml that you apply to all charts. The chart being applied only consumes part of it.

But, right now, users have a distinct values.yml per chart, and as you are saying, we are not checking for conflicts between them. So, users cannot merge them into 1 `my-values.yml`.

**Comment by ereslibre** [comment]
> But, right now, users have a distinct values.yml per chart, and as you are saying, we are not checking for conflicts between them. So, users cannot merge them into 1 my-values.yml.

We are in control of our helm chart values. We can avoid conflicts on our side. Automating this would be nice, but is not a must. Our Helm charts are not so complicated, specially `crd` and `defaults`.

With this approach, the complete `values.yaml` is not repeated three times (one per chart) in https://charts.kubewarden.io/, when in reality two charts don't use most of those values.

I think forcing one static `values.yaml` for all charts is too much and might confuse the knowledgeable user that wants to tweak values, having to check where values are used, because the `values.yaml` that comes with the chart is of no help (it just contains everything all charts support).

**Comment by ereslibre** [comment]
Also, just to add on top of that: given we ensure that there will be no conflicts, users can deploy the three charts applying the same `values.yaml` file to all of them. This is something we ensure will work.

**Comment by viccuad** [comment]
> I think forcing one static values.yaml for all charts is too much and might confuse the knowledgeable user that wants to tweak values, having to check where values are used,

Still, each charts ships the `chart-values.yml` just there for informative reasons. Users could have a look there for that.

> Also, just to add on top of that: given we ensure that there will be no conflicts, users can deploy the three charts applying the same values.yaml file to all of them.

But they need to construct that values.yml on their own.
I keep thinking that all charts are tightly coupled, and there's no value in deploying only 1. Hence, all users are going to need to construct that merged values on their own. Why not do that for them, and provide a simplex UX from the get go?

**Comment by ereslibre** [comment]
I updated the PR @viccuad so that it auto generates an `all-values.yaml` file that we can link from our documentation, so that users can just `wget` it if they want. I think that makes things easier while still not putting all values everywhere.

**Comment by viccuad** [comment]
Thanks for doing so. Still, that is not discoverable for users. I would prefer to just ship those `all-values.yml` in all charts. I will let others chime in.

**Review by raulcabello — APPROVED** [review]
LGTM, thanks!

**Pull request merged**